publisher-sdk/src/main/java/com/criteo/publisher/CriteoInterstitialActivity.java

```diff
@@ -36,26 +36,39 @@
 import androidx.annotation.VisibleForTesting;
 import com.criteo.publisher.adview.AdWebViewClient;
 import com.criteo.publisher.adview.RedirectionListener;
+import com.criteo.publisher.logging.Logger;
+import com.criteo.publisher.logging.LoggerFactory;
 import java.lang.ref.WeakReference;
 
 public class CriteoInterstitialActivity extends Activity {
 
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+
   private WebView webView;
   private ResultReceiver resultReceiver;
   private FrameLayout adLayout;
   private ComponentName callingActivityName;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
+    try {
+      super.onCreate(savedInstanceState);
+      doOnCreate();
+    } catch (Throwable t) {
+      logger.error("Error while creating interstitial activity.", t);
```

Comment:
I find it a bit risky to use the `LoggerFactory` here, because if we ever decide to inject the context when building a `LoggerFactory` then it would crash again. Could you add a comment on the `LoggerFactory` to explain that injecting a context could cause a side effect on `CriteoInterstitialActivity`?

Reply:
The test is checking this integration already and if someone make the logger factory crash, then it this test would be red.

Knowing this, do you confirm that you still want to add a comment ?

Reply:
In this case we are good! :)